### PR TITLE
Adding GitHub Action for CodeQL scanning

### PR DESCRIPTION
**Pull request by aprantl**
Adds [CodeQL](https://codeql.github.com/docs/codeql-overview/) scanning as GitHub Action.

False-Positive matches and similar can be suppressed in JavaScript/TypeScript files like:

```
function same(x, y)  {
    if (x !== x && y !== y)  // lgtm[js/comparison-of-identical-expressions]
        return true;
    return x === y;
}
```

(see [Alert suppression - Help - LGTM](https://lgtm.com/help/lgtm/alert-suppression#javascript))

### CODE REVIEW

**Comment by aprantl** [comment]
@mmichaelis @pkliesch What do you think should we maybe enable this by default. Kind of Sonar for this repository. It even found an issue :)

**Comment by mmichaelis** [comment]
Thanks, @aprantl, really looks good. Please, apply the following suppression to the example code alert, so that we don't start with a triggered alert:

```
const entitiesTableRow = (entity) => `<tr><td>${htmlCode(entity.replace("&", "&amp;"))}</td><td>${entity}</td></tr>`; // lgtm[js/incomplete-sanitization]
```

File: app/src/example-data.js [hunk 1]
```diff
@@ -1276,7 +1276,7 @@
   "&zwj;",
   "&zwnj;",
 ];
-const entitiesTableRow = (entity) => `<tr><td>${htmlCode(entity.replace("&", "&amp;"))}</td><td>${entity}</td></tr>`;
+const entitiesTableRow = (entity) => `<tr><td>${htmlCode(entity.replace("&", "&amp;"))}</td><td>${entity}</td></tr>`; // lgtm[js/incomplete-sanitization]
```
[the remark this answers is not in the document]
Comment (aprantl) [reply app/src/example-data.js:1279]: @mmichaelis - I was surprised to still see the alert. But apparently this is intentional - https://lgtm.com/help/lgtm/automated-code-review-faqs#why-automated-code-review-reports-misses-aler

[the remark this answers is not in the document]
Comment (mmichaelis) [reply app/src/example-data.js:1279]: Interesting. Good intention I think, that they don't want to let PRs silently introduce such suppression.

**Review by mmichaelis — APPROVED** [review]

**Pull request merged**